src/server/transaction.h

```diff
@@ -399,7 +400,7 @@ class Transaction {
     MultiMode mode;
 
     std::optional<IntentLock::Mode> lock_mode;
-    absl::flat_hash_set<std::string> locks;
+    std::vector<std::string> keys;  // List of all keys used when scheduling
 
```

Comment:
TODO: introduce compact string list type that will store a single buffer and fixed_array<uint16_t> sizes